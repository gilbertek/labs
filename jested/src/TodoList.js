import React, { Component} from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { addTodo, removeTodo } from './store';
import styles from './TodoList.css';

class TodoList extends Component {
  state = { input: '' };

  handleClick = i => () => {
    this.props.removeTodo(i);
  }

  handleChange = e => {
    this.setState({ input: e.currentTarget.value });
  }

  handleSubmit = () => {
    this.props.addTodo({ text: this.state.input });
    this.setState({ input: '' });
  }

  render() {
    return (
      <div className='todos--container'>
        <h1>Todos</h1>

        <input
          className='todos--addTodo'
          type='text'
          onChange={this.handleChange}
          value={this.state.input} />

        <ul>
          {this.props.todos.map(({ text }, i) => (
            <li onClick={this.handleClick(i)} key={i}>{text}</li>
          ))}
        </ul>

        <button
          className='todo--button'
          onClick={this.handleSubmit}>Add Todo</button>
      </div>
    );
  }
}

TodoList.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
};

export { TodoList };

const mapStateToProps = ({ currentList: { todos }}) => ({ todos });

const mapDispatchToProps = dispatch => {
  return {
    addTodo: text => dispatch(addTodo(text)),
    removeTodo: id => dispatch(removeTodo(id))
  };
}

const TodoListContainer = connect(mapStateToProps, mapDispatchToProps)(TodoList);
export default TodoListContainer;
