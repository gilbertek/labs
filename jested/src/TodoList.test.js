import React from 'react';
import ReactDOM from 'react-dom';
import { TodoList } from './TodoList';
import { configure, shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

configure({ adapter: new Adapter() })

const updateInput = ( wrapper, instance, newValue ) => {
  const input = wrapper.find(instance);
  input.simulate('change', {
    currentTarget: { value: newValue }
  });

  return wrapper.find(instance);
};

describe('<TodoList />', () => {
  it('calls addTodo Redux action creator with button click', () => {
    const props = {
      addTodo: jest.fn(),
      todos: []
    };

    const wrapper = shallow(<TodoList {...props} />);
    wrapper.find('.todos--addTodo').simulate('change', {
      currentTarget: { value: 'Buy eggs' }
    });
    wrapper.find('.todo--button').simulate('click');
    expect(props.addTodo).toHaveBeenCalledWith({ text: 'Buy eggs' });
  });

  it('calls removeTodo Redux action creator with button click', () => {
    const props = {
      removeTodo: jest.fn(),
      todos: [{ text: 'Buy eggs' }, { text: 'Change Oil' }]
    };

    const wrapper = shallow(<TodoList {...props} />);
    wrapper.find('li').at(0).simulate('click');
    expect(props.removeTodo).toHaveBeenCalledWith(0);
  });

  it('matches the snapshot', () => {
    const props = {
      addTodo: jest.fn(),
      todos: []
    };

    const tree = shallow(<TodoList {...props} />);
    expect(toJson(tree)).toMatchSnapshot();
  });
});
