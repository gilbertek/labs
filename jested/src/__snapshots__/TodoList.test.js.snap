// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TodoList /> matches the snapshot 1`] = `
<div
  className="todos--container"
>
  <h1>
    Todos
  </h1>
  <input
    className="todos--addTodo"
    onChange={[Function]}
    type="text"
    value=""
  />
  <ul />
  <button
    className="todo--button"
    onClick={[Function]}
  >
    Add Todo
  </button>
</div>
`;
